aws/cf_template.go

```diff
@@ -120,28 +125,41 @@ func generateTemplate(spec *stackSpec) (string, error) {
 			Description: "DNS name for the LoadBalancer",
 			Value:       cloudformation.GetAtt("LB", "DNSName").String(),
 		},
+		outputLoadBalancerARN: &cloudformation.Output{
+			Description: "ARN of the LoadBalancer",
+			Value:       cloudformation.Ref("LB").String(),
+		},
 		outputTargetGroupARN: &cloudformation.Output{
-			Description: "The ARN of the TargetGroup",
+			Description: "The ARN of the main TargetGroup",
 			Value:       cloudformation.Ref(httpsTargetGroupName).String(),
 		},
 	}
 
-	template.AddResource(httpsTargetGroupName, newTargetGroup(spec, parameterTargetGroupTargetPortParameter))
+	listener := ExtraListener{}
+	if spec.cascade {
+		listener = ExtraListener{ListenProtocol: "TCP", ListenPort: 443, TargetPort: 443, cascade: true}
+	}
+	template.AddResource(httpsTargetGroupName, newTargetGroup(spec, parameterTargetGroupTargetPortParameter, listener))
```

Comment:
can we do the code path with less changes on current CF stacks?
I would imagine it would be fine like:
```
extraListener := ExtraListener{}
if spec.cascade {
    listener = ExtraListener{ListenProtocol: "TCP", ListenPort: 443, TargetPort: 443, cascade: true}
    template.AddResource(httpsTargetGroupName, newTargetGroup(spec, parameterTargetGroupTargetPortParameter, extraListener))
} else {
    template.AddResource(httpsTargetGroupName, newTargetGroup(spec, parameterTargetGroupTargetPortParameter))
}
```
and later only use extraListener in case we have `if spec.cascade`.

Reply:
I think you still have to pass in the parameter to the function? Which is why it's just initialized as empty first before the if statement.